src/securechip/securechip.c

```diff
@@ -484,13 +484,25 @@ static ATCA_STATUS _update_kdf_key(void)
     UTIL_CLEANUP_32(encryption_key);
     _interface_functions->get_encryption_key(encryption_key);
 
+    uint8_t nonce_contribution[32] = {0};
+    UTIL_CLEANUP_32(nonce_contribution);
```

Comment:
Why do we clean this up if we're filling it just below?

Reply:
THE TIL_CLEANUP_ macros mark a stack var to be zeroed when the var goes out of scope, e.g. when the function returns, using the ` __attribute__((__cleanup__())` extension.

Reply:
TIL `TIL_CLEANUP`